startup/GafferCyclesUI/metadata.py

```diff
@@ -0,0 +1,923 @@
+             "brick_texture" }
+    
+    for x in node :
+        if plug.node()["name"].getValue() == x :
+            return plug.node()["parameters"][param].getValue() == value 
+
+def metadata( plug, name ) :
+
+    global parameterMetadata
+
+    shaderName = plug.node()["name"].getValue()
+    shaderDict = parameterMetadata.get( shaderName )
+    if shaderDict is None :
+        return None
+
+    parameterDict = shaderDict.get( plug.getName() )
+    if parameterDict is None :
+        return None
+    return parameterDict.get( name )
+
+### CyclesShader ###
+
+### main metadata assignments ###
+
+for name in ( "label", "layout:section", "layout:index", "userDefault" ) :
+    Gaffer.Metadata.registerValue( GafferCycles.CyclesShader, "parameters.*", name, functools.partial( metadata, name = name ) )
+
+### parameter visibility in principled_hair_bsdf and vector_rotate ### 
+
+Gaffer.Metadata.registerValue( GafferCycles.CyclesShader, "parameters.absorption_coefficient", "layout:visibilityActivator", functools.partial( visibility, value = "Absorption coefficient", param = "parametrization" ) )
+Gaffer.Metadata.registerValue( GafferCycles.CyclesShader, "parameters.melanin", "layout:visibilityActivator", functools.partial( visibility, value = "Melanin concentration", param = "parametrization" ) )
+Gaffer.Metadata.registerValue( GafferCycles.CyclesShader, "parameters.melanin_redness", "layout:visibilityActivator", functools.partial( visibility, value = "Melanin concentration", param = "parametrization" ) )
+Gaffer.Metadata.registerValue( GafferCycles.CyclesShader, "parameters.tint", "layout:visibilityActivator", functools.partial( visibility, value = "Melanin concentration", param = "parametrization" ) )
+Gaffer.Metadata.registerValue( GafferCycles.CyclesShader, "parameters.random_color", "layout:visibilityActivator", functools.partial( visibility, value = "Melanin concentration", param = "parametrization" ) )
+Gaffer.Metadata.registerValue( GafferCycles.CyclesShader, "parameters.color", "layout:visibilityActivator", functools.partial( visibility, value = "Direct coloring", param = "parametrization" ) )
+
+Gaffer.Metadata.registerValue( GafferCycles.CyclesShader, "parameters.rotation", "layout:visibilityActivator", functools.partial( visibility, value = "euler_xyz", param = "rotate_type" ) )
+Gaffer.Metadata.registerValue( GafferCycles.CyclesShader, "parameters.axis", "layout:visibilityActivator", functools.partial( visibility, value = "axis", param = "rotate_type" ) )
+
+mapping = [ "parameters.tex_mapping__translation", "parameters.tex_mapping__rotation", "parameters.tex_mapping__scale", "parameters.tex_mapping__use_minmax", 
+           "parameters.tex_mapping__min", "parameters.tex_mapping__max", "parameters.tex_mapping__x_mapping", "parameters.tex_mapping__y_mapping", 
+           "parameters.tex_mapping__z_mapping", "parameters.tex_mapping__type", "parameters.tex_mapping__projection" ]
```

Comment:
Formatting nitpick : please close braces at the same indentation level as they were opened at.

```
mapping = [ 
    ...
]
```